mapwfs.cpp

```diff
@@ -2540,6 +2540,9 @@ this request. Check wfs/ows_enable_request settings.", "msWFSGetFeature()",
         layerObj *lp;
         lp = GET_LAYER(map, j);
         if (lp->status == MS_ON) {
+          // classes don't affect retrieval of features, so set the count to 0
+          // this avoids selecting any fields referenced in a CLASS from the data source
+          lp->numclasses = 0;
```

Comment:
Won't that cause memory leaks when the lp object is cleaned up later ? (do we have msautotest testing that scenario by the way? if not would be appropriate to edit an existing one) It would probably be safer to save lp->numclasses values before setting it to 0 and restoring the original value after the msWFSRunBasicGetFeature() call.

I assume this could also be done before the call to FLTApplyFilterToLayer() done at line 2167.
mapogcapi.cpp could also benefit from that.

Reply:
Thanks for the review @rouault - much appreciated. 
There were no memory leaks as the layer cleanup code uses the `layer->maxclasses` value to loop through classes in the layer and free them rather than `lp->numclasses`. However it does seem safer to set `numclasses` back to the original value once the query is run. 

I've also added the same approach to `FLTApplyFilterToLayer` which also calls `buildLayerItemList`.